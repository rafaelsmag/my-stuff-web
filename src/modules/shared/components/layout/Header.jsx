import React from 'react'
import { Input, Image, Grid } from 'semantic-ui-react'
import Images from '../../../../config/constants/images'

const Header = (props) => (
  <Grid columns={3} centered>
    <Grid.Column only='tablet computer' floated='left' tablet={4} computer={4}>
      <Image src={Images.logo} size='tiny'/>
    </Grid.Column>
    <Grid.Column mobile={10} tablet={8} computer={8} textAlign='center' stretched verticalAlign='middle'>
      <Input placeholder='Search...' />
    </Grid.Column>
    <Grid.Column textAlign='right' mobile={6} tablet={4} computer={4}>
      <Image floated='right' src={Images.logo} size='tiny'/>
    </Grid.Column>
  </Grid>
)

export default Header
