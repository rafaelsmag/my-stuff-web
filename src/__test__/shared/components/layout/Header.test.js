import React from 'react'
import { shallow } from 'enzyme'

import Header from '../../../../modules/shared/components/layout/Header'

describe('Header Component', () => {
  const header = shallow(<Header />)
  it('should redenders properly', () => {
    expect(header).toMatchSnapshot()
  })
})
