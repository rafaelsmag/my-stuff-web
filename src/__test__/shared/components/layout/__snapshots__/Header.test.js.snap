// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component should redenders properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "centered": true,
      "children": Array [
        <GridColumn
          computer={4}
          floated="left"
          only="tablet computer"
          tablet={4}
>
          <Image
                    as="img"
                    size="tiny"
                    src="logo.svg"
                    ui={true}
          />
</GridColumn>,
        <GridColumn
          computer={8}
          mobile={10}
          stretched={true}
          tablet={8}
          textAlign="center"
          verticalAlign="middle"
>
          <Input
                    placeholder="Search..."
                    type="text"
          />
</GridColumn>,
        <GridColumn
          computer={4}
          mobile={6}
          tablet={4}
          textAlign="right"
/>,
      ],
      "columns": 3,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <Image
            as="img"
            size="tiny"
            src="logo.svg"
            ui={true}
/>,
          "computer": 4,
          "floated": "left",
          "only": "tablet computer",
          "tablet": 4,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "as": "img",
            "size": "tiny",
            "src": "logo.svg",
            "ui": true,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <Input
            placeholder="Search..."
            type="text"
/>,
          "computer": 8,
          "mobile": 10,
          "stretched": true,
          "tablet": 8,
          "textAlign": "center",
          "verticalAlign": "middle",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "placeholder": "Search...",
            "type": "text",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "computer": 4,
          "mobile": 6,
          "tablet": 4,
          "textAlign": "right",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "centered": true,
        "children": Array [
          <GridColumn
            computer={4}
            floated="left"
            only="tablet computer"
            tablet={4}
>
            <Image
                        as="img"
                        size="tiny"
                        src="logo.svg"
                        ui={true}
            />
</GridColumn>,
          <GridColumn
            computer={8}
            mobile={10}
            stretched={true}
            tablet={8}
            textAlign="center"
            verticalAlign="middle"
>
            <Input
                        placeholder="Search..."
                        type="text"
            />
</GridColumn>,
          <GridColumn
            computer={4}
            mobile={6}
            tablet={4}
            textAlign="right"
/>,
        ],
        "columns": 3,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <Image
              as="img"
              size="tiny"
              src="logo.svg"
              ui={true}
/>,
            "computer": 4,
            "floated": "left",
            "only": "tablet computer",
            "tablet": 4,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "as": "img",
              "size": "tiny",
              "src": "logo.svg",
              "ui": true,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <Input
              placeholder="Search..."
              type="text"
/>,
            "computer": 8,
            "mobile": 10,
            "stretched": true,
            "tablet": 8,
            "textAlign": "center",
            "verticalAlign": "middle",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "placeholder": "Search...",
              "type": "text",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "computer": 4,
            "mobile": 6,
            "tablet": 4,
            "textAlign": "right",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
